src/test/java/org/apache/commons/collections4/bloomfilter/hasher/HasherBuilderTest.java

```diff
@@ -108,4 +110,156 @@ static String getExtendedString() {
+        }
+    }
+
+    /**
+     * Test that adding a double into the hasher works correctly
+     */
+    @Test
+    public void withDoubleTest() {
+        TestBuilder builder = new TestBuilder();
+        Double[] values = { Double.valueOf(0.0), Double.MAX_VALUE, Double.MIN_VALUE, Double.MIN_NORMAL,
+            Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY, Double.NaN };
+        for (double d : values) {
+            builder.with(d);
+        }
+
+        for (int i = 0; i < values.length; i++) {
+            Assert.assertArrayEquals(ByteBuffer.allocate(Double.BYTES).putDouble(values[i]).array(),
+                    builder.items.get(i));
+        }
+    }
+
+    /**
+     * Test that adding a float into the hasher works correctly
+     */
+    @Test
+    public void withFloatTest() {
+        TestBuilder builder = new TestBuilder();
+        Float[] values = { Float.valueOf(0.0f), Float.MAX_VALUE, Float.MIN_VALUE, Float.MIN_NORMAL,
+            Float.NEGATIVE_INFINITY, Float.POSITIVE_INFINITY, Float.NaN };
+        for (float f : values) {
+            builder.with(f);
+        }
+
+        for (int i = 0; i < values.length; i++) {
+            Assert.assertArrayEquals(ByteBuffer.allocate(Float.BYTES).putFloat(values[i]).array(),
+                    builder.items.get(i));
+        }
+    }
+
+    /**
+     * Test that adding a character into the hasher works correctly
+     */
+    @Test
+    public void withCharTest() {
+        TestBuilder builder = new TestBuilder();
+        Character[] values = { Character.valueOf((char) 0), Character.MAX_HIGH_SURROGATE, Character.MAX_LOW_SURROGATE,
+            Character.MAX_SURROGATE, Character.MAX_VALUE, Character.MIN_HIGH_SURROGATE, Character.MIN_LOW_SURROGATE,
+            Character.MIN_SURROGATE, Character.MIN_VALUE };
+        for (char c : values) {
+            builder.with(c);
+        }
+
+        for (int i = 0; i < values.length; i++) {
+            Assert.assertArrayEquals(ByteBuffer.allocate(Character.BYTES).putChar(values[i]).array(),
+                    builder.items.get(i));
+        }
+    }
+
+    /**
+     * Test that adding a short into the hasher works correctly
+     */
+    @Test
+    public void withShortTest() {
+        TestBuilder builder = new TestBuilder();
+        Short[] values = { Short.valueOf((short) 0), Short.MAX_VALUE, Short.MIN_VALUE };
+        for (short s : values) {
+            builder.with(s);
+        }
+
+        for (int i = 0; i < values.length; i++) {
+            Assert.assertArrayEquals(ByteBuffer.allocate(Short.BYTES).putShort(values[i]).array(),
+                    builder.items.get(i));
+        }
+    }
+
+    /**
+     * Test that adding a BigInteger into the hasher works correctly
+     */
+    @Test
+    public void withBigIngeterTest() {
+        TestBuilder builder = new TestBuilder();
+        BigInteger[] values = { BigInteger.ZERO, BigInteger.ONE, BigInteger.TWO, BigInteger.TEN,
```

Comment:
@Claudenw it looks like `.TWO` is private in Java 8, only `ONE, `TEN`, `ZERO` are available: https://docs.oracle.com/javase/8/docs/api/java/math/BigInteger.html

There's an error in the Java 8 build in Travis.